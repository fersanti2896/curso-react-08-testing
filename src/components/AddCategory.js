import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => { /* Recibe a las categorias */
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault() /* Esto previene que se refresque la pagina */
        if(inputValue.trim().length > 2) {
            setCategories(cat => [inputValue, ...cat]); /* Comunicacion entre componentes */
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    );
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
