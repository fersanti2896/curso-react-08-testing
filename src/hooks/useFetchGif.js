import { useEffect, useState } from "react"
import { getGifs } from '../helpers/getGifs'

export const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data: [], 
        loading: true
    });

    useEffect(() => {
        /* Se importa la funciÃ³n que llama a la API y devuelve una promesa */
        getGifs(category)
            .then(imgs => setState({
                data: imgs,
                loading: false
            }))
    }, [category]);
    
    return state;
}