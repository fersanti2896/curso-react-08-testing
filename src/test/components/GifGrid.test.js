import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif');

describe('Haciendo pruebas en componente GifGrid', () => {
    const category = 'Dr. House';

    test('Debe mostrar <GifGrid /> correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan las imÃ¡genes useFetchGifs', () => {
      /* Usamos un mock  */
        const gifs = [{
            id: 'ABC',
            url: 'https:/localhost/algo/algo.png', 
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
});