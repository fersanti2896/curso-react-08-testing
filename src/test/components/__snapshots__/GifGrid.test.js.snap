// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haciendo pruebas en componente GifGrid Debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeftBig"
  >
    Dr. House
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Haciendo pruebas en componente GifGrid Debe mostrar items cuando se cargan las im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeftBig"
  >
    Dr. House
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https:/localhost/algo/algo.png"
    />
  </div>
</Fragment>
`;
